FreeRTOS/Test/CMock/event_groups/event_groups_utest.c

```diff
@@ -0,0 +1,608 @@
+ * https://github.com/FreeRTOS
+ *
+ */
+/*! @file list_utest.c */
+
+/* C runtime includes. */
+#include <stdlib.h>
+#include <stdbool.h>
+
+/* List includes */
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "event_groups.h"
+
+/* Test includes. */
+#include "unity.h"
+#include "unity_memory.h"
+
+/* Mock includes. */
+#include "mock_task.h"
+#include "mock_timers.h"
+#include "mock_list.h"
+#include "mock_list_macros.h"
+#include "mock_fake_assert.h"
+#include "mock_fake_port.h"
+
+
+/* ===========================  DEFINES CONSTANTS  ========================== */
+#define BIT_0            ( 1 << 0 )
+#define BIT_2            ( 1 << 2 )
+#define BIT_4            ( 1 << 4 )
+#define ALL_SYNC_BITS    ( BIT_0 | BIT_2 | BIT_4 )
+
+/* ===========================  GLOBAL VARIABLES  =========================== */
+
+/**
+ * @brief Global message buffer handle used for tests.
+ */
+static EventGroupHandle_t xEventGroupHandle;
+static List_t xListTemp = { 0 };
+static List_t * pxListTemp = &xListTemp;
+static ListItem_t xListItemDummy = { 0 };
+static ListItem_t * pxListItem_HasTaskBlockOnBit0 = &xListItemDummy;
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+
+/* ==========================  CALLBACK FUNCTIONS =========================== */
+
+void * pvPortMalloc( size_t xSize )
+{
+    return unity_malloc( xSize );
+}
+void vPortFree( void * pv )
+{
+    return unity_free( pv );
+}
+
+/* ============================  Unity Fixtures  ============================ */
+/*! called before each testcase */
+void setUp( void )
+{
+    xEventGroupHandle = NULL;
+
+    pxListTemp->pxIndex = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->xListEnd.xItemValue = portMAX_DELAY;
+    pxListTemp->xListEnd.pxNext = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->xListEnd.pxPrevious = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->uxNumberOfItems = ( UBaseType_t ) 0U;
+
+    pxListItem_HasTaskBlockOnBit0->xItemValue = BIT_0;
+
+    vFakeAssert_Ignore();
+    vFakePortEnterCriticalSection_Ignore();
+    vFakePortExitCriticalSection_Ignore();
+    ulFakePortSetInterruptMaskFromISR_IgnoreAndReturn( 0U );
+    vFakePortClearInterruptMaskFromISR_Ignore();
+
+    /* Track calls to malloc / free */
+    UnityMalloc_StartTest();
+}
+
+/*! called after each testcase */
+void tearDown( void )
+{
+    UnityMalloc_EndTest();
+}
+
+/*! called at the beginning of the whole suite */
+void suiteSetUp()
+{
+}
+
+/*! called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ===========================  Static Functions  =========================== */
+
+
+
+/* ==============================  Test Cases  ============================== */
+
+/*!
+ * @brief validate dynamically creating and deleting a new RTOS event group,
+ * @coverage xEventGroupCreate
+ */
+void test_xEventGroupDynamicCreateAndDelete_Success( void )
```

Comment:
You're missing a coverage tag for vEventGroupDelete, so it is being exluded.

Reply:
Done